### PR TITLE
Improved existing phone number verification jquery hack

### PR DESCRIPTION
**Pull request by Gotham13121997**
#191 , #196

### CODE REVIEW

File: mainapp/templates/mainapp/contributor_form.html [hunk 1]
```diff
@@ -23,7 +23,13 @@
   </form>
 <script>
     window.onload = function() {
-        $("#id_phone").attr('type', 'number');
+        $("#id_phone").attr({'type':'tel',
+                'pattern':'[0-9]{10}'}).keypress(function (e) {
+     //if the letter is not digit then display error and don't type anything
+     if (e.which != 8 && e.which != 0 && (e.which < 48 || e.which > 57)) {
+               return false;
```
Comment (salmanulfarzy) [inline mainapp/templates/mainapp/contributor_form.html]: It would help users if we respond with some response instead of nothing.

Comment (Gotham13121997) [reply mainapp/templates/mainapp/contributor_form.html]: Character key can be pressed only in desktops or laptops. On mobile devices it should switch to Numeric keypad automatically. I dont think a prompt is necessary.

**Comment by biswaz** [comment]
@vrthra Requesting testing

File: mainapp/templates/mainapp/contributor_form.html (hunk 1 above)
Comment (salmanulfarzy) [reply mainapp/templates/mainapp/contributor_form.html]: No response while typing could leave the user on confusion whether his machine or connection has stalled. If we're ignoring user input, It's always better to acknowledge it to the user.

Comment (Gotham13121997) [reply mainapp/templates/mainapp/contributor_form.html]: hmmm I understand. Then what do you propose . I dont think an alert will be a good idea. And anything else can hamper the ui.

Comment (Gotham13121997) [reply mainapp/templates/mainapp/contributor_form.html]: Any way there should be backend verification as well. We can never rely on front end validation

**Comment by vrthra** [comment]
Tested on the desktop.

**Comment by Gotham13121997** [comment]
@vrthra sure . On it

**Comment by Gotham13121997** [comment]
@vrthra Please test

**Comment by vrthra** [comment]
@Gotham13121997 
Ok, here is what I did; I tried adding 11 clicked submit. It told me that there is an error. Then I continued with 22 33 44 55 at each point it told me to provide a valid phone number. The error stayed even after I reached all ten digits (55), and went away only after a few seconds. This can cause confusion. Is there a better way?

**Comment by Gotham13121997** [comment]
ok I will check

**Comment by shajith** [comment]
@Gotham13121997 didn't seem to stop me from submitting "123" as a phone number on iOS Safari

**Comment by Gotham13121997** [comment]
@shajith thats weird. Were you able to submit the form?? This is why backend verification is necessary.

**Comment by Gotham13121997** [comment]
@shajith it seems oninvalid is not supported in safari

**Comment by shajith** [comment]
@Gotham13121997 caniuse indicates that iOS Safari supports input field patterns, and it works on http://html5pattern.com/. The other validations kick in, just the pattern thing - it let me submit with just "123". It did prevent submission if I left the field blank though.

**Comment by Gotham13121997** [comment]
@shajith can you try this commit and check. If not working then the default message will have to be left as it is

**Comment by shajith** [comment]
sorry for the lag, but no luck - same behavior on latest.

**Comment by Gotham13121997** [comment]
We will need to rely on server side verification for safari as it seems pattern is not supported. @shajith  @biswaz @vrthra @salmanulfarzy #196

File: mainapp/templates/mainapp/contributor_form.html (hunk 1 above)
Comment (Neonkraft) [reply mainapp/templates/mainapp/contributor_form.html]: I have a fix for this in #484

**Comment by Gotham13121997** [comment]
closing this as #484  is better option

**Pull request closed without merging**